src/index/CompressedBlockPrefiltering.h

```diff
@@ -127,17 +154,36 @@ class NotExpression : public PrefilterExpression {
   std::unique_ptr<PrefilterExpression> child_;
 
  public:
-  explicit NotExpression(std::unique_ptr<PrefilterExpression> child)
-      : child_(child->logicalComplement()) {}
+  explicit NotExpression(std::unique_ptr<PrefilterExpression> child,
+                         bool makeCopy = false)
```

Comment:
In the case where you call `makeCopy`, why can't you then use the child directly instead of having this argument here with strange semantics?
I'll have a look at it.